app/ui/src/app/api/providers/api-http-provider.service.ts

```diff
@@ -150,7 +151,7 @@ export class ApiHttpProviderService extends ApiHttpService {
       .filter(requestEvent => requestEvent.type === HttpEventType.Response)
       .map(requestEvent => requestEvent as HttpResponse<T>)
       .map(requestEvent => requestEvent.body)
-      .catch(error => Observable.throw(this.catchError(JSON.parse(error.error))));
+      .catch(error => this.handleError(error.error));
```

Comment:
I'd suggest passing the whole error object here too...

Reply:
This is basically the change I had in #2342 let's just go with this PR for that.

Reply:
Problem is that, contrary to the other `httpClient` handlers, the object returned through the RxJS `catch` handler here is not the actual HTTP error, but a native Angular wrapper around it, typed as a `HttpErrorResponse` object. Therefore the properties will not match with those expected by the logic in `handleError()`.

More info: https://github.com/angular/angular/blob/5.2.9/packages/common/http/src/response.ts#L288-L329

Reply:
ah, okay gotcha didn't realize that, thanks!  Well then I'm all +1 on this PR then :-)